tools/src/main/java/org/apache/kafka/trogdor/workload/PayloadGeneratorManager.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.trogdor.workload;
+
+/**
+ * Maintains the position number for a PayloadGenerator.
+ */
+public final class PayloadGeneratorManager {
```

Comment:
I like the approach of separating the current position and the iteration from the generator, but the name `Manager` does not seem very suggestive of its use. Perhaps `PayloadIterator` or something like that would be clearer?

Reply:
Yeah, `PayloadIterator` is a great name.  Will change.